File: xiaocheng-netdisk/mobile/lib/data/models/share.dart
class Share {
  final Long id;
  final String shareCode;
  final Long userId;
  final String fileIds;
  final int shareType;
  final String? password;
  final DateTime? expireTime;
  final int viewCount;
  final int downloadCount;
  final int status;
  final DateTime createTime;

  Share({
    required this.id,
    required this.shareCode,
    required this.userId,
    required this.fileIds,
    required this.shareType,
    this.password,
    this.expireTime,
    required this.viewCount,
    required this.downloadCount,
    required this.status,
    required this.createTime,
  });

  factory Share.fromJson(Map<String, dynamic> json) {
    return Share(
      id: json['id'],
      shareCode: json['shareCode'],
      userId: json['userId'],
      fileIds: json['fileIds'],
      shareType: json['shareType'],
      password: json['password'],
      expireTime: json['expireTime'] != null ? DateTime.parse(json['expireTime']) : null,
      viewCount: json['viewCount'],
      downloadCount: json['downloadCount'],
      status: json['status'],
      createTime: DateTime.parse(json['createTime']),
    );
  }

  String get shareUrl => 'xiaocheng://share/$shareCode';

  String get shareTypeDesc {
    switch (shareType) {
      case 1:
        return '公开链接';
      case 2:
        return '密码链接';
      case 3:
        return '指定好友';
      default:
        return '未知';
    }
  }

  bool get isExpired => expireTime != null && expireTime!.isBefore(DateTime.now());
}

File: xiaocheng-netdisk/backend/src/main/java/com/xiaocheng/netdisk/dal/entity/Share.java
package com.xiaocheng.netdisk.dal.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.util.Date;

@Data
@TableName("share")
public class Share {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String shareCode;

    private Long userId;

    private String fileIds;

    private Integer shareType;

    private String password;

    private Date expireTime;

    private Integer viewCount;

    private Integer downloadCount;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private Date createTime;
}
